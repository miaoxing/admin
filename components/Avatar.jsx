import {Avatar as AntdAvatar} from 'antd';
import propTypes from 'prop-types';
import defaultAvatar from '../images/avatar.jpg';

const Avatar = ({user, ...rest}) => {
  return (
    <AntdAvatar src={user.avatar || defaultAvatar} {...rest}/>
  );
};

Avatar.propTypes = {
  user: propTypes.shape({
    avatar: propTypes.string,
  }),
};

export default Avatar;
