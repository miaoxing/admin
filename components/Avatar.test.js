import Avatar from './Avatar';
import {render} from '@testing-library/react';

describe('Avatar', () => {
  test('default', () => {
    const {container} = render(<Avatar user={{}}/>);

    expect(container).toMatchSnapshot();
  });

  test('custom', () => {
    const {container} = render(<Avatar user={{avatar: 'abc.png'}}/>);

    expect(container).toMatchSnapshot();
  });
});
