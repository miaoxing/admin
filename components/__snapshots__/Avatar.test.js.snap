// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar custom 1`] = `
<div>
  <span
    class="ant-avatar ant-avatar-circle ant-avatar-image"
  >
    <img
      src="abc.png"
    />
  </span>
</div>
`;

exports[`Avatar default 1`] = `
<div>
  <span
    class="ant-avatar ant-avatar-circle ant-avatar-image"
  >
    <img
      src="test-file-stub"
    />
  </span>
</div>
`;
